flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -682,6 +683,9 @@ private ApplicationReport startAppMaster(
 				// add user code jars from the provided JobGraph
 				: jobGraph.getUserJars().stream().map(f -> f.toUri()).map(File::new).collect(Collectors.toSet());
 
+		int fileReplication = yarnConfiguration.getInt(DFSConfigKeys.DFS_REPLICATION_KEY, DFSConfigKeys.DFS_REPLICATION_DEFAULT);
+		fileReplication = flinkConfiguration.getInteger(YarnConfigOptions.FILE_REPLICATION, fileReplication);
```

Comment:
I think we need a test with configuration setting to `-1` (we currently only have one test to set to `4`)
my gut feeling is that this won't work - Configuration.getInteger only applies default value when the config key is `null`, however the default value according to the documentation is `-1` not `null`.

Reply:
Yes, you are right. I changed the logic to if the configured value is less than default value -1. It will fall back to use the yarn replication configuration. The logic is tested in another test case in YarnITCase.